### DISCUSSION TITLE
Application access expiration for hybrid SPAPIn app vs MWS access?

### DISCUSSION THREAD

**Discussion by IgniteMosquito** [amzn/selling-partner-api-models#3176] [General]
Hello,
We've noticed that the new app access grant for Hybrid SPAPI app expires at the same time as MWS access.
There are two confusing things, which I would like to understand and can't find in the documentation:
1. If the seller authorizes our SPAPI app access, why expiration date is the same as old MWS? In some cases, user will have to do the same steps a few days later.
2. When we re-enable MWS access, this doesn't trigger for SPAPI. Is it a desired behaviour, that we need to repeat authorization flow? Seller doesn't have "Renew" button in "Manage your Apps" tab for SPAPI app.

And is there a plan to extend the token life, so we do not need to go through the full authorization flow again every year?

**Comment by quynhkim** [comment]
I would like the same thing: NO "Renew" SP-token on the users' end. If users don't want to use any app, they just need to disable that app, that's it. 

Please do not make SP-token automatically expire (after 1 year as it is now) because users are using the app and suddenly SP-API return error and it really confuses them and make the life the the app developer extremely difficult.
